src/deploy/functions/runtimes/node/parseTriggers.ts

```diff
@@ -173,6 +202,146 @@ export function mergeRequiredAPIs(backend: backend.Backend) {
   backend.requiredAPIs = merged;
 }
 
+/**
+ *
+ */
+export function addResourcesToBuild(
+  projectId: string,
+  runtime: runtimes.Runtime,
+  annotation: TriggerAnnotation,
+  want: build.Build
+): void {
+  Object.freeze(annotation);
+  // for (const region of annotation.regions || [api.functionsDefaultRegion]) {
+  const regions = annotation.regions || [api.functionsDefaultRegion];
+  let triggered: build.Triggered;
+
+  const triggerCount =
+    +!!annotation.httpsTrigger +
+    +!!annotation.eventTrigger +
+    +!!annotation.taskQueueTrigger +
+    +!!annotation.blockingTrigger;
+  if (triggerCount !== 1) {
+    throw new FirebaseError(
+      "Unexpected annotation generated by the Firebase Functions SDK. This should never happen."
+    );
+  }
+
+  if (annotation.taskQueueTrigger) {
+    want.requiredAPIs.push({
+      api: "cloudtasks.googleapis.com",
+      reason: "Needed for task queue functions.",
+    });
+    triggered = {
+      taskQueueTrigger: {},
+    };
+    proto.copyIfPresent(triggered.taskQueueTrigger, annotation.taskQueueTrigger, "invoker");
+    proto.copyIfPresent(triggered.taskQueueTrigger, annotation.taskQueueTrigger, "rateLimits");
+    if (annotation.taskQueueTrigger.retryConfig) {
+      triggered.taskQueueTrigger.retryConfig = Object.assign(
+        annotation.taskQueueTrigger.retryConfig,
+        { maxRetryDurationSeconds: 0 }
```

Comment:
why are you overriding maxRetryDuration to always be 0 here?

Reply:
It's not actually setting to 0, it's really awkwardly initializing it with a default that gets overridden later, but there's a much less dumb way to do this that I swapped to. Definitely worth pointing out the awkwardness.